File: server/model/sys.go
package model

import (
	"github.com/ethanwang9/covid19/server/global"
	"go.uber.org/zap"
)

// name: 系统信息表
// author: Ethan.Wang
// desc:

type Sys struct {
	Base
	Id        string `gorm:"colmun:id" json:"id,omitempty" `
	Copyright string `gorm:"colmun:copyright" json:"copyright"`
	GovNo1    string `gorm:"colmun:gov_no1" json:"gov_no1"`
	GovNo2    string `gorm:"colmun:gov_no2" json:"gov_no2"`
	MpUrl     string `gorm:"colmun:mp_url" json:"mp_url"`
	MpImg     string `gorm:"colmun:mp_img" json:"mp_img"`
	Mail      string `gorm:"colmun:mail" json:"mail"`
	Blog      string `gorm:"colmun:blog" json:"blog"`
}

var SysApp = new(Sys)

func (s *Sys) New(sys Sys) *Sys {
	sys.Id = "9"
	return &sys
}

// Get 获取系统信息
func (s *Sys) Get() (Sys, error) {
	var t Sys
	err := global.DB.Where("id = ?", s.Id).Find(&t).Error
	if err != nil {
		global.LOG.Error("GORM|系统信息表#获取系统信息失败", zap.Error(err))
		return Sys{}, err
	}

	return t, nil
}

// Set 设置系统信息
func (s *Sys) Set() error {
	err := global.DB.Model(&Sys{}).Where("id = ?", s.Id).Updates(map[string]interface{}{
		"copyright": s.Copyright,
		"gov_no1":   s.GovNo1,
		"gov_no2":   s.GovNo2,
		"mp_url":    s.MpUrl,
		"mp_img":    s.MpImg,
		"mail":      s.Mail,
		"blog":      s.Blog,
	}).Error
	if err != nil {
		global.LOG.Error("GORM|系统信息表#设置系统信息失败", zap.Error(err), zap.Any("data", s))
		return err
	}

	return nil
}

File: web/src/api/sys.ts
// name: 系统信息接口
// author: Ethan.Wang
// desc:

import {PromiseRes} from "./api";
import request from "./index";

// 获取微信公众号信息 - 返回
export interface SysMpInfoRes {
    appid: string
    secret: string
}

// 获取微信公众号信息
export const SysMpInfo = (): PromiseRes<SysMpInfoRes> => request({
    method: "get",
    params: {},
    url: "/sys/mp",
})

// ===================================

// 设置微信公众号信息 - 请求
export interface SysMpInfoSetReq {
    appid: string
    secret: string
}

// 设置微信公众号信息
export const SysMpInfoSet = (data: SysMpInfoSetReq): PromiseRes<null> => request({
    method: "post",
    data,
    url: "/sys/mp",
})

// ===================================

// 获取系统信息 - 返回
export interface SysInfoRes {
    copyright: string
    gov_no1: string
    gov_no2: string
    mp_url: string
    mp_img: string
    mail: string
    blog: string
}

// 获取系统信息
export const SysInfo = (): PromiseRes<SysInfoRes> => request({
    method: "get",
    params: {},
    url: "/sys/info",
})

// ===================================

// 设置系统信息 - 请求
export interface SysInfoSetReq {
    copyright: string
    gov_no1: string
    gov_no2: string
    mp_url: string
    mp_img: string
    mail: string
    blog: string
}

// 设置系统信息
export const SysInfoSet = (data: SysInfoSetReq): PromiseRes<null> => request({
    method: "post",
    data,
    url: "/sys/info",
})